app/controllers/concerns/api/v3/sync_to_user.rb

```diff
@@ -87,9 +87,10 @@ def resync_token_modified?
     end
 
     def sync_region_modified?
-      return if requested_sync_region_id.blank?
+      # If the user has been syncing a different region than
+      # what we think the user's sync region should be, we resync.
       return if process_token[:sync_region_id].blank?
-      process_token[:sync_region_id] != requested_sync_region_id
+      process_token[:sync_region_id] != current_sync_region.id
```

Comment:
The main fix happens by setting the appropriate `sync_region_id` in the facility's payload. The app can then ask us for the right sync region and all should be well.

This change is due to an edge case we came across when a user is enabled as an MO for the first time.
1. A user is syncing block data as a regular user. Their sync header is set to block ID.
2. They are enabled as a teleconsultation MO
3. If they sync patients, BPs before resyncing facilities, they will end up in a broken state because of this check ^.